### PR TITLE
Add `Application.fetch_env!/2` and simplify `Application.get_env/3`

### PR DESCRIPTION
**Pull request by lexmag**

### CODE REVIEW

File: lib/elixir/lib/application.ex [hunk 1]
```diff
@@ -148,6 +145,23 @@ defmodule Application do
...
+  If the specified application is not loaded, or the configuration parameter
+  does not exist, raises `ArgumentError`.
+  """
+  @spec fetch_env!(app, key) :: value | no_return
+  def fetch_env!(app, key) do
+    case fetch_env(app, key) do
+      {:ok, value} -> value
+      :error ->
+        raise ArgumentError,
+          "application #{inspect app} is not loaded, " <>
```
Comment (josevalim) [inline lib/elixir/lib/application.ex]: This is not strictly correct because you can have configuration without the application being loaded, can't you?

Comment (lexmag) [inline lib/elixir/lib/application.ex]: It took it verbatim from the `fetch_env/2` doc. :bowtie:
It depend on what "loaded" means, if it means "started" you are right; but I assume it means something like "that app is not in dependencies". Anyway it has to be clarified, do you have any suggestion?

Comment (lexmag) [inline lib/elixir/lib/application.ex]: Actually it comes from Erlang docs: http://www.erlang.org/doc/apps/kernel/application.html#get_env-3.

Comment (josevalim) [inline lib/elixir/lib/application.ex]: Ok, so let's ship it.

Comment (lexmag) [inline lib/elixir/lib/application.ex]: \o/ can we have it backported to v1.0?

Comment (lexmag) [inline lib/elixir/lib/application.ex]: Ignore me, it's not a bug.

**Pull request merged**